lua/treesitter-context.lua

```diff
@@ -82,27 +47,48 @@ local function can_open(bufnr, winid)
   return true
 end
 
-local update = throttle(function()
```

Comment:
Throttle is needed to prevent the context from updating too often when the user is scrolling. Otherwise it leads to lag.

Reply:
What's the problem in leaving it for all events? We need to throttle for each window. You can find a modified version of this function in Gitsigns that allows functions with different arguments to be throttled independently. That can probably be used here.